src/internal/m365/collection/drive/restore_caches.go

```diff
@@ -21,6 +22,11 @@ type driveInfo struct {
 	rootFolderID string
 }
 
+type AvailableEntities struct {
```

Comment:
Prefer the term `Resource` over `Entity`, please.  Entity is too generic, and can reference non-resource entries.

```suggestion
type ResourceIDNames struct {
```